handler/engine/src/test/java/io/knotx/fragments/engine/Nodes.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2019 Knot.x Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.knotx.fragments.engine;
+
+import static io.knotx.fragments.handler.api.domain.FragmentResult.ERROR_TRANSITION;
+import static io.knotx.fragments.handler.api.domain.FragmentResult.SUCCESS_TRANSITION;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.matches;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import io.knotx.fragments.engine.graph.CompositeNode;
+import io.knotx.fragments.engine.graph.Node;
+import io.knotx.fragments.engine.graph.NodeType;
+import io.knotx.fragments.engine.graph.SingleNode;
+import io.knotx.fragments.handler.api.domain.FragmentContext;
+import io.knotx.fragments.handler.api.domain.FragmentResult;
+import io.reactivex.Single;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+
+interface Nodes {
+
+  static CompositeNode composite(String nodeId, List<Node> rootNodes) {
+    CompositeNode node = mock(CompositeNode.class);
+    when(node.getId()).thenReturn(nodeId);
+    when(node.getType()).thenReturn(NodeType.COMPOSITE);
+    when(node.getNodes()).thenReturn(rootNodes);
+    return node;
+  }
+
+  static CompositeNode composite(String nodeId, List<Node> rootNodes, Node success) {
+    CompositeNode node = composite(nodeId, rootNodes);
+    when(node.next(matches(SUCCESS_TRANSITION))).thenReturn(Optional.ofNullable(success));
+    return node;
+  }
+
+  static CompositeNode composite(String nodeId, List<Node> rootNodes, Node success, Node error) {
+    CompositeNode node = composite(nodeId, rootNodes, success);
+    when(node.next(matches(ERROR_TRANSITION))).thenReturn(Optional.ofNullable(error));
+    return node;
+  }
+
+  static SingleNode node(String nodeId, Function<FragmentContext, Single<FragmentResult>> function) {
```

Comment:
To be consistent: `node` -> `single` ?

Reply:
Sure, we can change this value. However, the `SingleNode` name is the best one - when you check the interface you will see that this node allows modifying a Fragment. It is more like `FunctionNode` or `ProcessingNode`. `CompositeNode` is a special node that exposes its internal complexity to Engine. But still, we can implement `SingleNode` that would work like `CompositeNode`.

Reply:
Ok, I was not precise. I meant method name.

Reply:
Fixed, however still `single` is not good name:
```
interface SingleNode extends Node {

  Single<FragmentResult> execute(FragmentContext fragmentContext);

}
```